Library/ViewModels/PledgeViewModel.swift

```diff
@@ -74,13 +97,174 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
     self.shippingLocationViewHidden = reward
       .map { $0.shipping.enabled }
       .negate()
+
+    self.configureStripeIntegration = projectAndReward.ignoreValues()
+      .map { _ in
+        (
+          PKPaymentAuthorizationViewController.merchantIdentifier,
+          AppEnvironment.current.environmentType.stripePublishableKey
+        )
+      }
+
+    let selectedShippingRule = Signal.merge(
+      projectAndReward.mapConst(nil),
+      self.shippingRuleSelectedSignal.wrapInOptional()
+    )
+
+    let backingData: Signal<BackingData, Never> = Signal.combineLatest(
+      projectAndReward,
+      pledgeAmount,
+      selectedShippingRule
+    )
+    .map { projectAndReward, pledgeAmount, selectedShippingRule in
+      (
+        projectAndReward.0,
+        projectAndReward.1,
+        pledgeAmount,
+        selectedShippingRule
+      )
+    }
+
+    // Apple Pay
+
+    let paymentAuthorizationData = backingData
+      .map { (
+        $0.project,
+        $0.reward,
+        $0.pledgeAmount,
+        $0.selectedShippingRule,
+        PKPaymentAuthorizationViewController.merchantIdentifier
+      ) as PaymentAuthorizationData }
+
+    self.goToApplePayPaymentAuthorization = paymentAuthorizationData
+      .takeWhen(self.applePayButtonTappedSignal)
+
+    let stripeToken: Signal<String, Never> = self.tokenOrErrorSignal
```

Comment:
For simplicity seeing as we only care about this being `.left(token)` we could lose the `switch` and do:

```swift
let stripeToken: Signal<String, Never> = self.tokenOrErrorSignal
  .map { tokenOrError -> String? in
    if case let .left(token) = tokenOrError {
      return token
    }

    return nil
  }
  .skipNil()
```